game/src/core/Game.java

```diff
@@ -316,24 +506,17 @@ private void updateSystems() {
     public void onLevelLoad() {
         currentLevel = levelManager.getCurrentLevel();
         removeAllEntities();
-        getHero().ifPresent(this::placeOnLevelStart);
-        getHero().ifPresent(Game::addEntity);
-        try {
-            EntityFactory.getChest();
-        } catch (IOException e) {
-            // will be moved to MAIN in https://github.com/Programmiermethoden/Dungeon/pull/688
-            LOGGER.warning("Could not create new Chest: " + e.getMessage());
-            throw new RuntimeException();
-        }
+        hero().ifPresent(this::placeOnLevelStart);
+        hero().ifPresent(Game::addEntity);
+        userOnLevelLoad.execute();
     }
 
     /** Set the focus of the camera on the hero, if he exists otherwise focus on Pont (0,0) */
     private void setCameraFocus() {
-        if (getHero().isPresent()) {
+        if (hero().isPresent()) {
```

Comment:
@AMatutat Den Codeblock
```java
        if (hero().isPresent()) {
            PositionComponent pc =
                    (PositionComponent)
                            hero()
                                    .get()
                                    .getComponent(PositionComponent.class)
                                    .orElseThrow(
                                            () ->
                                                    new MissingComponentException(
                                                            "PositionComponent"));
            camera.setFocusPoint(pc.getPosition());

        } else camera.setFocusPoint(new Point(0, 0));
```

könnte man (lesbarer) durch
```java
        Supplier<MissingComponentException> mce = () -> new MissingComponentException("PositionComponent");
        Function<Entity, PositionComponent> getPositionComponent = e -> (PositionComponent) e.getComponent(PositionComponent.class).orElseThrow(mce);
        hero().ifPresentOrElse(
            e -> camera.setFocusPoint(getPositionComponent.apply(e).getPosition()),
            () -> camera.setFocusPoint(new Point(0, 0)));
```

ersetzen.

Reply:
Oder mit zwei neuen Hilfsmethoden:

```java
    private void setCameraFocus() {
        getHero().ifPresentOrElse(
            Game::setFocusPointToEntity,
            Game::setFocusPointToZeroZero
        );
    }

    private static void setFocusPointToZeroZero() {
        camera.setFocusPoint(new Point(0, 0));
    }

    private static void setFocusPointToEntity(Entity e) {
        Supplier<MissingComponentException> mce = () -> new MissingComponentException("PositionComponent");
        Function<Entity, PositionComponent> getPositionComponent = e -> (PositionComponent) e.getComponent(PositionComponent.class).orElseThrow(mce);
        camera.setFocusPoint(getPositionComponent.apply(e).getPosition());
    }
```

Reply:
Genau genommen gehören die Hilfsmethoden aber nicht zu `Game`, sondern in den Typ der `camera`.

Reply:
ist ein schöner vorschlag/einwand, wird mit #699 aber eh überflüssig. 
Aber ich habs mal auf den Schirm, für andere Konstrukte dieser Art.